### PR TITLE
Fix incorrect and no support for background image calculations of 16 and 32 bit radiographs

### LINKED ISSUE #244: setBackground - mayo shoulder data crash
Tested from #76

The mayo radiograph test set (shldimg0005  Prop_Int)  I have are incredibly 'dense' black when rendered. I wanted to see if setBackground could provide a better input to the ncc  calc

on setBackground (following threshold popup) a QWindowsWindow error is returned in cmd and and unhandled exception thrown line 186 Video.cpp

 
setBackground_QWindow
shoulder_drr_radDARK

With the Knee data (311135_L_Hop) the same QWindowWindows error message is populated in cmd but the unhandled exception does not occur
setBackground_KNEEQWindow

### PR DESCRIPTION
**Pull request by NicerNewerCar**
* Closes #244

### CODE REVIEW

File: libautoscoper/src/Video.cpp [hunk 1]
```diff
@@ -194,6 +186,18 @@ int Video::create_background_image()
   return 1;
 }
 
+template <typename T> void Video::create_background_image_internal(TiffImage* tmp_img, unsigned int normalization_factor) {
```
Comment (NicerNewerCar) [inline libautoscoper/src/Video.cpp:189]:
```suggestion
template <typename T> void Video::create_background_image_internal(TiffImage* tmp_img, unsigned int normalization_factor) {
```

Any suggestions for a better name for this function?

**Comment by jcfr** [comment]
In the case of `unsigned int`, we also have the following ...

image

We should account for this.

**Comment by jcfr** [comment]
@NicerNewerCar 

Could you suggest a commit message to use for `Squash & Merge` ? Also if the PR effectively fixes the incorrect computation that was associated with 8-bit images, what was fixed should be explained. 

My understanding is that the PR is generalizing the existing (and correct) 8-bit background computation to 16 and 32-bit radiographs.

**Comment by jcfr** [comment]
Is the fix `COMP: Fix multiple min/max macros on windows` required in this pull request ? It seems this should be contributed it its own pull request ?

**Review by jcfr — COMMENTED** [review]
See above

**Comment by NicerNewerCar** [comment]
> Is the fix `COMP: Fix multiple min/max macros on windows` required in this pull request ? It seems this should be contributed it its own pull request ?

Yes, the following line will not work on Windows without that commit. 

```
  constexpr unsigned int normalization_factor = std::numeric_limits<T>::max();
```

**Review by jcfr — APPROVED** [review]

**Pull request merged**